In [1]:
#importing necessary libraries
import numpy as np
import pandas as pd

In [2]:
#importing the dataset from amazon
df_movies = pd.read_csv('Amazon - Movies and TV Ratings.csv')

## Analysing the dataset

In [3]:
#viewing the shape of the dataset
df_movies.shape

(4848, 207)

In [5]:
df_movies.head(5)

user_id  Movie1  Movie2  Movie3  Movie4  Movie5  Movie6  Movie7  \
0  A3R5OBKS7OM2IR     5.0     5.0     NaN     NaN     NaN     NaN     NaN   
1   AH3QC2PC1VTGP     NaN     NaN     2.0     NaN     NaN     NaN     NaN   
2  A3LKP6WPMP9UKX     NaN     NaN     NaN     5.0     NaN     NaN     NaN   
3   AVIY68KEPQ5ZD     NaN     NaN     NaN     5.0     NaN     NaN     NaN   
4  A1CV1WROP5KTTW     NaN     NaN     NaN     NaN     5.0     NaN     NaN   

   Movie8  Movie9  ...  Movie197  Movie198  Movie199  Movie200  Movie201  \
0     NaN     NaN  ...       NaN       NaN       NaN       NaN       NaN   
1     NaN     NaN  ...       NaN       NaN       NaN       NaN       NaN   
2     NaN     NaN  ...       NaN       NaN       NaN       NaN       NaN   
3     NaN     NaN  ...       NaN       NaN       NaN       NaN       NaN   
4     NaN     NaN  ...       NaN       NaN       NaN       NaN       NaN   

   Movie202  Movie203  Movie204  Movie205  Movie206  
0       NaN       NaN       NaN       NaN       NaN  
1       NaN       NaN       NaN       NaN       NaN  
2       NaN       NaN       NaN       NaN       NaN  
3       NaN       NaN       NaN       NaN       NaN  
4       NaN       NaN       NaN       NaN       NaN  

[5 rows x 207 columns]

In [4]:
df_movies.columns

Index(['user_id', 'Movie1', 'Movie2', 'Movie3', 'Movie4', 'Movie5', 'Movie6',
       'Movie7', 'Movie8', 'Movie9',
       ...
       'Movie197', 'Movie198', 'Movie199', 'Movie200', 'Movie201', 'Movie202',
       'Movie203', 'Movie204', 'Movie205', 'Movie206'],
      dtype='object', length=207)

The dataset has movie reviews which are directly tagged to the user using the unique 'user_id'. Also all the movies are not watched by all users

In [11]:
df_movies['user_id'].isna().value_counts()

False    4848
Name: user_id, dtype: int64

None of the user ids are NA. So we have data for all 4848 user ids.

In [13]:
df_movies.isna().sum()

user_id        0
Movie1      4847
Movie2      4847
Movie3      4847
Movie4      4846
            ... 
Movie202    4842
Movie203    4847
Movie204    4840
Movie205    4813
Movie206    4835
Length: 207, dtype: int64

## Movies with maximum views/ ratings

In [16]:
df_movies['Movie192'].value_counts()

5.0    6
4.0    3
3.0    1
Name: Movie192, dtype: int64

## Average rating for each movie

## Top 5 movies by ratings

## Providing ratings for all movies for all users

## Model building and predictions